cbcontainers/state/components/enforcer_deployment.go

```diff
@@ -142,6 +142,7 @@ func (obj *EnforcerDeploymentK8sObject) mutateEnforcerEnvVars(container *coreV1.
 		{Name: "GUARDRAILS_ENFORCER_KEY_FILE_PATH", Value: fmt.Sprintf("%s/key", DesiredTlsSecretVolumeMountPath)},
 		{Name: "GUARDRAILS_ENFORCER_CERT_FILE_PATH", Value: fmt.Sprintf("%s/signed_cert", DesiredTlsSecretVolumeMountPath)},
 		{Name: "GUARDRAILS_ENFORCER_PROMETHEUS_PORT", Value: fmt.Sprintf("%d", enforcerSpec.Prometheus.Port)},
+		{Name: "GUARDRAILS_ENFORCER_PORT", Value: fmt.Sprintf("%d", DesiredContainerPortValue)},
```

Comment:
Note that using the `env` key should still override this if it is needed but without it we were relying on whatever is specified as `default value` in the `DataplaneConfig` struct, which means we had to keep it manually in sync with the port here - not ideal IMO.